artifacts/scripts/util.sh

```diff
@@ -243,6 +248,19 @@ sync_repo() {
         git commit -q -m "sync: remove kubernetes-sha"
     fi
 
+    # remove existing recursive-delete-pattern files. After a first removal commit, the filter-branch command
+    # will filter them out from upstream commits.
+    if [ -n "${recursive_delete_pattern}" ]; then
+        local split_recursive_delete_pattern
+        read -r -a split_recursive_delete_pattern <<< "${recursive_delete_pattern}"
+        git rm -q --ignore-unmatch -r "${split_recursive_delete_pattern[@]}"
+        git add -u
+        if ! git-index-clean; then
+            echo "Deleting files recursively: ${recursive_delete_pattern}"
+            git commit -q -m "sync: initially remove files ${recursive_delete_pattern}"
```

Comment:
can the commit list the files it deleted? (instead of pattern) (or list the files being deleted in log).

Reply:
That's one file per directory, and for each commit. That's spam the logs a lot.

Reply:
actually, here we can be verbose. It's just the initial commit. Done.